tempesta_fw/http_msg.c

```diff
@@ -42,18 +42,39 @@ tfw_http_msg_make_hdr(TfwPool *pool, const char *name, const char *val)
 {
 	size_t n_len = strlen(name);
 	size_t v_len = val ? strlen(val) : 0;
-	const TfwStr tmp_hdr = {
-		.chunks = (TfwStr []){
-			{ .data = (void *)name,	.len = n_len },
-			{ .data = (void *)val,	.len = v_len,
-			  .flags = TFW_STR_HDR_VALUE},
-		},
-		.len = n_len + v_len,
-		.eolen = 2,
-		.nchunks = (val ? 2 : 1)
-	};
+	size_t n;
+	TfwStr *hdr, *h_c;
+	char *data;
+
+	n = ((val ? 2 : 1) + 1) * sizeof(TfwStr) + n_len + v_len;
+	hdr = (TfwStr *)tfw_pool_alloc(pool, n);
+	if (!hdr)
+		return NULL;
+	TFW_STR_INIT(hdr);
 
-	return tfw_strdup(pool, &tmp_hdr);
+	hdr->len = n_len + v_len;
+	hdr->eolen = 2;
+	hdr->nchunks = (val ? 2 : 1);
+	hdr->chunks = hdr + 1;
+	data = (char *)(TFW_STR_LAST(hdr) + 1);
+
+	h_c = TFW_STR_CHUNK(hdr, 0);
+
+	TFW_STR_INIT(h_c);
+	h_c->data = data;
+	h_c->len = n_len;
+	tfw_cstrtolower_wo_avx2(data, name, n_len);
+
+	data += h_c->len;
+	++h_c;
+
+	TFW_STR_INIT(h_c);
```

Comment:
The reason why `resp_hdr_set|add` tests fails on your branch is here. The allocated memory segment is used like this:
`| compound str | sub str for name | sub str for value | name buf | value buf |` . But if the value is missing, the string becomes shorter: `| compound str | sub str for name | name buf | ` . 

In this code segment substring for value in unconditionally written to memory segment after name substring. But on short allocations the whole value substring is missing and name buffer is overwriten with TfwStr header.